ledger/participant-integration-api/src/test/suite/scala/platform/store/appendonlydao/events/TransactionsReaderSpec.scala

```diff
@@ -3,64 +3,191 @@
 
 package com.daml.platform.store.appendonlydao.events
 
+import org.scalacheck.{Gen, Shrink}
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
-private[appendonlydao] class TransactionsReaderSpec extends AnyWordSpec with Matchers {
+private[appendonlydao] class TransactionsReaderSpec
+    extends AnyWordSpec
+    with Matchers
+    with ScalaCheckPropertyChecks {
   "splitRange" should {
-    "correctly split in equal ranges" in {
-      TransactionsReader.splitRange(100L, 200L, 4, 10, 100) shouldBe Vector(
+    "correctly split in equal ranges by number of chunks" in {
+      TransactionsReader.splitRange(
+        startExclusive = 100L,
+        endInclusive = 200L,
+        numberOfChunks = 4,
+        maxChunkSize = 100,
+      ) shouldBe Vector(
         EventsRange(100L, 125L),
         EventsRange(125L, 150L),
         EventsRange(150L, 175L),
         EventsRange(175L, 200L),
       )
     }
 
-    "correctly split in non-equal ranges" in {
-      TransactionsReader.splitRange(100L, 200L, 3, 10, 100) shouldBe Vector(
-        EventsRange(100L, 134L),
-        EventsRange(134L, 168L),
-        EventsRange(168L, 200L),
+    "fairly split in non-equal ranges by number of chunks" in {
+      TransactionsReader.splitRange(
+        startExclusive = 100L,
+        endInclusive = 200L,
+        numberOfChunks = 6,
+        maxChunkSize = 100,
+      ) shouldBe Vector(
+        EventsRange(100L, 117L),
+        EventsRange(117L, 134L),
+        EventsRange(134L, 151L),
+        EventsRange(151L, 168L),
+        EventsRange(168L, 184L),
+        EventsRange(184L, 200L),
       )
     }
 
-    "output ranges of sizes at least minChunkSize" in {
-      TransactionsReader.splitRange(100L, 200L, 3, 50, 100) shouldBe Vector(
+    "output ranges of at most max chunk size" in {
+      TransactionsReader.splitRange(
+        startExclusive = 100L,
+        endInclusive = 200L,
+        numberOfChunks = 1,
+        maxChunkSize = 50,
+      ) shouldBe Vector(
         EventsRange(100L, 150L),
         EventsRange(150L, 200L),
       )
     }
 
-    "output one range if minChunkSize higher than range size" in {
-      TransactionsReader.splitRange(100L, 200L, 3, 200, 100) shouldBe Vector(
-        EventsRange(100L, 200L)
+    "output one range if use maxChunkSize/10 higher than range size" in {
```

Comment:
I don't understand this sentence

Reply:
Right, clarified with `output one range if maxChunkSize/10 (as minimum chunk size) is higher than range size`